src/routes/categories/containers/CategoriesContainer.js

```diff
@@ -40,66 +40,57 @@ type StatePropsType = StatusPropsType<ContainerPropsType, RefreshPropsType>
 type DispatchPropsType = {| dispatch: Dispatch<StoreActionType> |}
 type PropsType = {| ...OwnPropsType, ...StatePropsType, ...DispatchPropsType |}
 
-const createChangeUnavailableLanguage = (path: string, navigation: NavigationScreenProp<*>, city: string) => (
+const createChangeUnavailableLanguage = (city: string) => (
   dispatch: Dispatch<StoreActionType>, newLanguage: string
 ) => {
   const switchContentLanguage: SwitchContentLanguageActionType = {
     type: 'SWITCH_CONTENT_LANGUAGE',
     params: { newLanguage, city }
   }
   dispatch(switchContentLanguage)
-  const navigateToCategory = createNavigateToCategory('Categories', dispatch, navigation)
-  navigateToCategory({
-    cityCode: city,
-    language: newLanguage,
-    path,
-    forceUpdate: false,
-    key: navigation.state.key
-  })
 }
 
 const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsType => {
   if (!state.cityContent) {
     return { status: 'routeNotInitialized' }
   }
-  const { resourceCache, categoriesRouteMapping, switchingLanguage } = state.cityContent
+  const { resourceCache, categoriesRouteMapping, switchingLanguage, languages } = state.cityContent
   const route = categoriesRouteMapping[ownProps.navigation.state.key]
-
   if (!route) {
     return { status: 'routeNotInitialized' }
   }
 
-  const city = route.city
-  const refreshProps = { cityCode: city, language: route.language, path: route.path, navigation: ownProps.navigation }
-
-  if (state.cities.status === 'error' ||
-    resourceCache.errorMessage !== undefined ||
-    route.status === 'error') {
-    return { status: 'error', refreshProps }
-  }
-
-  if (route.status === 'loading' || switchingLanguage || state.cities.status === 'loading') {
+  if (state.cities.status === 'loading' || switchingLanguage || route.status === 'loading' || !languages) {
     return { status: 'loading' }
   }
 
-  const cities = state.cities.models
-  const languages = Array.from(route.allAvailableLanguages.keys())
-  const stateView = new CategoriesRouteStateView(route.path, route.models, route.children)
-  if (!languages.includes(route.language)) {
+  if (route.status === 'languageNotAvailable') {
     return {
       status: 'languageNotAvailable',
-      availableLanguages: languages,
-      cityCode: city,
-      refreshProps,
-      changeUnavailableLanguage: createChangeUnavailableLanguage(route.path, ownProps.navigation, city)
+      availableLanguages: languages.filter(lng => route.allAvailableLanguages.has(lng.code)),
```

Comment:
Why is this called `allAvailableLanguages` btw? Is there a reason for it? One could assume `availableLanguages` are already all available languages.

Reply:
It's an implicit indicator that it also contains the current language (in contrast to CategoryModel.availableLanguages)